File: selected_coding_interview/codes/java/lc_400_nth_digit/lc_400_nth_digit_s2.java
/*
* File: lc_400_nth_digit_s2.java
* Created Time: 2025-12-30
* Author: krahets
*/

package lc_400_nth_digit;

import include.*;
import java.util.*;

// ===== Solution Code =====
    long num = start + (n - 1) / digit;

public class lc_400_nth_digit_s2 {
    public static void main(String[] args) {
        // ======= Test Case =======
        int test_input = 11;
        int expected_output = 0;

        // ====== Driver Code ======
        Solution slt = new Solution();
        var result = slt.findNthDigit(test_input);
        System.out.println(result);

    }
}

File: selected_coding_interview/codes/python/lc_400_nth_digit_s2.py
"""
File: lc_400_nth_digit_s2.py
Created Time: 2025-12-30
Author: krahets
"""

from include import *


# ===== Solution Code =====
class Solution:
    def findNthDigit(self, n: int) -> int:
        digit, start, count = 1, 1, 9
        while n > count:
            n -= count
            start *= 10
            digit += 1
            count = 9 * start * digit
        num = start + (n - 1) // digit
        return int(str(num)[(n - 1) % digit])


# ======= Test Case =======
test_input = 11
expected_output = 0

# ====== Driver Code ======
slt = Solution()
result = slt.findNthDigit(test_input)
print(result)
